mapper.go

```diff
@@ -439,17 +439,17 @@ func mapDecoder(r *Registry) MapperFunc {
 			target.Set(reflect.MakeMap(target.Type()))
 		}
 		el := target.Type()
-		sep := ctx.Value.Tag.MapSep
+		mapsep := ctx.Value.Tag.MapSep
 		var childScanner *Scanner
 		if ctx.Value.Flag != nil {
 			t := ctx.Scan.Pop()
 			// If decoding a flag, we need an argument.
 			if t.IsEOL() {
-				return errors.New("unexpected EOL")
+				return fmt.Errorf("missing argument: expecting \"<key>=<value>%c...\"", mapsep)
```

Comment:
This is awesome thank you, just two minor things.

1. Please replace "argument" with "value", as "argument" has an existing, different, meaning.
2. Please change "argument: expecting" to "argument, expecting"